library/padlock.c

```diff
@@ -33,6 +33,10 @@
 
 #if defined(MBEDTLS_HAVE_X86)
 
+#if defined(MBEDTLS_AES_USE_HARDWARE_ONLY)
+#error "MBEDTLS_PADLOCK_C defined, but not all prerequisites"
```

Comment:
agreed - this is very definitely not "a prerequisite" missing

Reply:
I agree with @gilles-peskine-arm . That follows up similar error message.

Reply:
As far as I can see, "not all prerequisites" is used when we have `#if defined(FOO) && !defined(BAR)`, rather than an extra macro defined (the "all prerequisites" and be `!defined (BAR1) && !defined(BAR2) && ...`)

Reply:
> As far as I can see, "not all prerequisites" is used when we have `#if defined(FOO) && !defined(BAR)`, rather than an extra macro defined (the "all prerequisites" and be `!defined (BAR1) && !defined(BAR2) && ...`)

That's in `padlock.c`. It looks confuse others. I will move it.